tools/RunGenMain.cpp

```diff
@@ -499,6 +513,63 @@ int main(int argc, char **argv) {
         }
     }
 
+    if (skip_bad_environment) {
+        // Let's do some preflighting to see if we are trying to run on a system that
+        // definitely doesn't support our target. For now, we're just checking Cuda
+        // because it's the only thing needed for current Halide tests.
+        //
+        // TODO: add checking for other GPU (etc) backends.
+        std::set<std::string> tokens;
+        {
+            std::stringstream s(rf->filter_metadata->target);
+            std::string tok;
+            while (std::getline(s, tok, '-')) {
+                tokens.insert(tok);
+            }
+        }
+        if (tokens.count("cuda")) {
+            int version_required = 20;  // Minimum for any Halide-generated Cuda output
+            static const std::pair<const char *, int> capabilities[] = {
+                {"cuda_capability30", 30},
+                {"cuda_capability32", 32},
+                {"cuda_capability35", 35},
+                {"cuda_capability50", 50},
+                {"cuda_capability61", 61},
+                {"cuda_capability70", 70},
+                {"cuda_capability75", 75},
+                {"cuda_capability80", 80},
+                {"cuda_capability86", 86},
+            };
+            for (const auto &it : capabilities) {
+                if (tokens.count(it.first)) {
+                    version_required = std::max(version_required, it.second);
+                }
+            }
+
+            void *cuda_intf = halide_get_symbol("halide_cuda_device_interface");
```

Comment:
<nit> consider moving this check to the start of this ``if`` block so we exit early in the instances that we need to.